llvm/lib/CAS/BuiltinCAS.cpp

```diff
@@ -2687,20 +2687,28 @@ OnDiskCAS::getObjectProxy(IndexProxy I) const {
 
   // Helper for creating the return.
   auto createProxy = [&](MemoryBufferRef Buffer) -> ObjectProxy {
-    assert(Buffer.getBuffer().drop_back(Blob0).end()[0] == 0 &&
-           "Null termination");
-    if (Blob)
+    if (Blob) {
+      assert(Buffer.getBuffer().drop_back(Blob0).end()[0] == 0 &&
+             "Null termination should be guaranteed by creation logic");
```

Comment:
Maybe be more specific about the assertion message about blob is null terminated? Same for the one below.

Reply:
Cleaned up in 0a213bf584da7fe0aa660fbf7ac877544b3d2efe.